File: 1-50/problem10/problem10.py
primes = [2]
MAX = 2000000
n = 3
while n < MAX:
    m = n ** 0.5 + 1
    is_prime = True
    for p in primes:
        if p > m: break
        if n % p == 0:
            is_prime = False
            break
    if is_prime:
        primes.append(n)
    n += 2
print(sum(primes))

File: 1-50/problem10/problem10.scala
def step(start: Long, by: Long): Stream[Long] = start #:: step(start + by, by)
val primes: Stream[Long] = 2L #:: step(3L, 2).filter(x => primes.takeWhile(_ <= math.sqrt(x)).forall(x % _ != 0))

println(primes.takeWhile(_ < 2000000).sum) // 142913828922